File: 160-IntersectionOfTwoLinkedLists/160-IntersectionOfTwoLinkedLists.c
// Last updated: 6/6/2026, 10:26:36 PM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB) {
    struct ListNode *p=headA;
    struct ListNode *q=headB;
    while(p!=q)
    {
        if(p==NULL)
        {
            p=headB;
        }
        else
        {
            p=p->next;
        }

        if(q==NULL)
        {
            q=headA;
        }
        else
        {
            q=q->next;
        }

    }
    return p;
}

File: 160-IntersectionOfTwoLinkedLists/160-IntersectionOfTwoLinkedLists.py
# Last updated: 6/6/2026, 10:26:31 PM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        a = headA
        b = headB

        while a!=b:
            if a:
                a = a.next
            else:
                a = headB
            if b:
                b = b.next
            else:
                b = headA
        return b
        